doc/api/addons.md

```diff
@@ -1047,20 +1044,23 @@ The implementation in `myobject.cc` is similar to the previous example:
 
 namespace demo {
 
+using node::AddEnvironmentCleanupHook;
 using v8::Context;
 using v8::Function;
 using v8::FunctionCallbackInfo;
 using v8::FunctionTemplate;
+using v8::Global;
 using v8::Isolate;
 using v8::Local;
 using v8::NewStringType;
 using v8::Number;
 using v8::Object;
-using v8::Persistent;
 using v8::String;
 using v8::Value;
 
-Persistent<Function> MyObject::constructor;
+// Warning! This is not thread-safe, this addon cannot be used for worker
+// threads.
```

Comment:
Not a big fan of docs suggesting to write code in a way that's not thread-safe, even if it _does_ warn about it. Would it be particularly difficult to change these examples to be thread-safe? :thinking:

Reply:
> Would it be particularly difficult to change these examples to be thread-safe? thinking

Yes, sadly the point of these examples is to show a specific pattern which doesn’t easily allow thread-safe usage.

I didn’t want to expand the scope of this PR that far, but a major cleanup of the addon docs is long overdue anyway, and I’d like to follow up with that.